app/controllers/publishers_controller.rb

```diff
@@ -17,7 +17,12 @@ class PublishersController < ApplicationController
   ].freeze
 
   before_action :authenticate_via_token, only: %i(show)
-  before_action :authenticate_publisher!, except: %i(show)
+  before_action :authenticate_publisher!, except: %i(
+    show
+    two_factor_authentication_removal
```

Comment:
In my mind this is kind of clear indication that these methods likely belongs in another controller. This controller relates to a publisher and almost all of these methods require authentication via publisher. 

The reasoning I have for moving it into a new controller is
* they don't require any publisher to be logged in
* relates to a 2FA "model" as in everything about these can be mapped to create, and update